### PR TITLE
CLE-NST-AE - Add tests for Eithers

### PR DESCRIPTION
**Pull request by IgorWolkov**

### CODE REVIEW

File: src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala [hunk 1]
```diff
@@ -218,8 +218,14 @@ object macros:
...
 
+        case '[Left[t, f]] =>
+          Json.fromFields(("Left", deconstructArgument[t]) :: Nil)
+
+        case '[Right[t, f]] =>
+          Json.fromFields(("Right", deconstructArgument[f]) :: Nil)
+
         case '[Either[t, f]] =>
-          Json.fromFields((StringUnit, deconstructArgument[f]) :: Nil)
+          Json.fromFields(("Right", deconstructArgument[f]) :: Nil)
```
Comment (VLADISLAV008) [inline src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala]: This implementation does not handle every possible case.
In the following case, there will be no compilation error:
```
        extension (inline sc: StringContext)
            inline def encode(inline args: Any*): Json =
               ${ macros.encode[Either[Int, List[String]]]('sc, 'args) }
              
        val value: Either[String, List[String]] = Left("")
           
          encode"""$value """
```

**Review by VLADISLAV008 — CHANGES_REQUESTED** [review]
I left a comment.

File: src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala (hunk 1 above)
Comment (IgorWolkov) [reply src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala]: Seems we have an issue in tests and this case must be added.

**Review by VLADISLAV008 — APPROVED** [review]
LGTM

**Review by mikhailZakharevich — APPROVED** [review]

File: src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala [hunk 2]
```diff
@@ -724,7 +743,13 @@ object macros:
...
+
+    def generateFreshKey(stringContextParts: List[String]): String =
+      val key = UUID.randomUUID().toString
+      if stringContextParts exists { str => str contains key} then generateFreshKey(stringContextParts)
+      else key
+
+
   end encode
 
 end macros
```
Comment (mikhailZakharevich) [inline src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala:755]: needs an empty line at the eof

File: src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala [hunk 3]
```diff
@@ -65,9 +71,15 @@ object macros:
...
-          ScalaTry(validateJsonSchema[T]("*", jsonSchema.hcursor)) match {
+
+          val stringContextParts = getStringContextParts(sc)
+
+          val eitherLeftFreshKey: EitherLeftFreshKey = EitherLeftFreshKey(generateFreshKey(stringContextParts))
+          val eitherRightFreshKey: EitherRightFreshKey = EitherRightFreshKey(generateFreshKey(stringContextParts))
+
+          val jsonSchema = makeJsonSchema(stringContextParts, deconstructArguments(eitherLeftFreshKey, eitherRightFreshKey)(argExprs))
+
+          ScalaTry(validateJsonSchema[T]("*", jsonSchema.hcursor)(using eitherLeftFreshKey, eitherRightFreshKey)) match {
```
Comment (mikhailZakharevich) [inline src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala:82]: Do you want to pass eitherLeftFreshKey, eitherRightFreshKey explicitly?

**Pull request merged**